docs/1-introduction/1-1-getting-started/1-1-2-hello-d3d11.md

```diff
@@ -181,7 +181,21 @@ if (FAILED(CreateDXGIFactory2(
 
 The first part calls the parent class, where `GLFW` is initialized and setup.
 
-!!! error "What is `IID_PPV_ARGS`"?
+`IID_PPV_ARGS(type)` Is a complex compile time wrapper that (at a basic level) expands the inputted pointer into `_uuidof(*type), (void **)(type)` which means that for functions that have a `REFIID` as a parameter and immediatly after a `[out] void**` parameter, this macro will expand the IID_PPV_ARGS(type) expression into these parameters — this is seen with `CreateDXGIFactory2` where the second last and last parameter are a `REFIID` and `void**`:
+```cpp
+HRESULT CreateDXGIFactory2(
+        UINT   Flags,
+        REFIID riid,
+  [out] void   **ppFactory
+);
+```
+What this expression does as a result is 2 different things for the 2 things it expands into:
+
+[the `_uuidof(type)` part of `IID_PPV_ARGS(type)`]
+-  Computes and returns at compile time a Reference (REF) to an Interface Identifier (IID) (hence the type name of REFIID) based on the type of variable passed as an argument. A (REF)IID is a type of GUID (globally unique identifier), which means in simple terms that it is just an arbitrary value which always maps directly to the same variable type each and every time called.
```

Comment:
It does not "compute", should be: "`__uuidof(type)` retrieves the UUID of this MIDL type. MIDL, or Microsoft Interface Definition Language is a particular convention in use by Microsoft where every type has a UUID attached to it, since D3D11 uses MIDL every ID3D11XXX type has UUIDs"